dsl/src/commonMain/kotlin/it/unibo/collektive/networking/Message.kt

```diff
@@ -13,13 +13,62 @@ sealed interface Message
 data class InboundMessage<ID : Any>(val senderId: ID, val messages: Map<Path, *>) : Message
 
 /**
- * An [OutboundMessage] are [messages] that a device [senderId] sends to all other neighbors.
+ * An [OutboundMessage] are messages that a device [senderId] sends to all other neighbours.
  */
-data class OutboundMessage<ID : Any>(val senderId: ID, val messages: Map<Path, SingleOutboundMessage<ID, *>>) : Message
+class OutboundMessage<ID : Any>(
+    expectedSize: Int,
+    val senderId: ID,
+) : Message {
+
+    /**
+     * The default messages to be sent to all neighbours.
+     */
+    val defaults: MutableMap<Path, Any?> = LinkedHashMap(expectedSize * 2)
+    private val overrides: MutableMap<ID, MutableList<Pair<Path, Any?>>> = LinkedHashMap(expectedSize * 2)
+
+    /**
+     * Check if the [OutboundMessage] is empty.
+     */
+    fun isEmpty(): Boolean = defaults.isEmpty()
+
+    /**
+     * Check if the [OutboundMessage] is not empty.
+     */
+    fun isNotEmpty(): Boolean = defaults.isNotEmpty()
```

Comment:
Do we really need both functions?

Reply:
I'd keep them both in coherence with Kotlin's stdlib